### DISCUSSION TITLE
第3节课笔记 (15班)

### DISCUSSION THREAD

**Discussion by szhowardhuang** [InternLM/Tutorial#160] [笔记]
按照LangChain教程，
1） 安装环境，过程顺利，huggingface 下载 sentence transformer，要用镜像站点。model可以存在之前创建的/root/model目录里面。
![image](https://github.com/InternLM/tutorial/assets/1407300/f9e8a58e-207a-4592-9b6d-685470065d40)
2) 代码下载， 可下载到code目录，不过之后脚本或者代码要注意目录
![image](https://github.com/InternLM/tutorial/assets/1407300/45d4572d-a19c-44b6-96f3-9ce5004a5c73)
3）数据收集， 建立data目录 /root/data ， 把数据集放到此目录.  把脚本拷贝到data目录， cp -r /root/code/tutorial/langchain/demo/ ./
![image](https://github.com/InternLM/tutorial/assets/1407300/3e638c5a-379b-4196-a3fc-58d221281487)
4）构建向量数据库， 执行create_db.py即可。注意调整model路径， 数据库文件会存到当前python执行路径
5）部署web demo，由于我用VSC remote，懒得改隧道端口。所以更改一下教程里面的server port。

![image](https://github.com/InternLM/tutorial/assets/1407300/cbc63a8c-94b9-4664-b655-f511cdc0b7d0)

执行刚开始要等一下
![image](https://github.com/InternLM/tutorial/assets/1407300/2131bca1-c9ee-4578-a1bb-262aa77f1927)

效果如下， 类似的问题得到不同的结果，看起来要用中文切片模型。
![image](https://github.com/InternLM/tutorial/assets/1407300/3f3d84ae-9c91-4298-af30-ae490f9a65c4)

为了确认是否切片模型造成乱码，下载bge-large-zh
import os
os.environ['HF_ENDPOINT'] = 'https://hf-mirror.com'
os.system('huggingface-cli download --resume-download BAAI/bge-large-zh --local-dir /root/model/bge-large-zh')

修改构建向量数据库模型和查询向量数据库模型，记得删除之前建立的数据库。
embeddings = HuggingFaceEmbeddings(model_name="/root/model/bge-large-zh")

效果如下
![image](https://github.com/InternLM/tutorial/assets/1407300/f309f387-7f57-4392-8f5f-189cd9ba5c52)

**Comment by seienn** [comment]
https://gvxxiiwsufy.feishu.cn/wiki/NCClweMjkiAwxwkXXcOcqAV2nXe?from=from_copylink
